File: 1/2020.10.27/6道股票问题/714 手续费/java/solution.java
class Solution {
    public int maxProfit(int[] prices, int fee) {
        int dp0 = 0, dp1 = Integer.MIN_VALUE;
        for (int price : prices) {
            dp0 = Math.max(dp0, dp1 + price);
            dp1 = Math.max(dp1, dp0 - price - fee);
        }
        return dp0;
    }
}

File: 1/2020.10.27/6道股票问题/714 手续费/python/solution.py
class Solution:
    def maxProfit(self, prices: List[int], fee: int) -> int:
        dp0, dp1 = 0, float('-inf')
        for price in prices:
            dp0 = max(dp0, dp1 + price)
            dp1 = max(dp1, dp0 - price - fee)
        return dp0
